### PR TITLE
Remove unneeded changes from Skia

### PR DESCRIPTION
**Pull request by kaidokert**
Cleans up some unneeded no-op changes.
    
b/304784019

### CODE REVIEW

File: third_party/skia/include/core/SkTypes.h [hunk 1]
```diff
@@ -653,4 +638,19 @@ enum class SkBackingFit {
...
+        return SkPmcolorIsRgba;
+    } else if (SK_PMCOLOR_BYTE_ORDER(B,G,R,A)) {
+        return SkPmcolorIsBgra;
+    } else {
+        SkASSERTF(false,"SK shift values do not correspond to a supported byte order.");
+        return (SkPmcolor)NULL;
+    }
+}
+#endif
+
```
Comment (dahlstrom-g) [inline third_party/skia/include/core/SkTypes.h:655]: Why is this moved?

File: third_party/skia/src/core/SkShaderCodeDictionary.h [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 #include <array>
 #include <unordered_map>
+#include <string>
```
Comment (dahlstrom-g) [inline third_party/skia/src/core/SkShaderCodeDictionary.h:13]: Why is this added?

File: third_party/skia/include/core/SkTypes.h (hunk 1 above)
Comment (kaidokert) [reply third_party/skia/include/core/SkTypes.h:655]: It uses skASSERTF which is defined earlier in this file. Skia code does not use DCHECK/CHECKs, so our earlier modification needed an update to avoid pulling in //base/logging

File: third_party/skia/src/core/SkShaderCodeDictionary.h (hunk 2 above)
Comment (kaidokert) [reply third_party/skia/src/core/SkShaderCodeDictionary.h:13]: Because the file uses `std::string` and doesn't have an include for it. This single line would be an upstreamable patch, assuming upstream trunk is still similar.

File: third_party/skia/src/sksl/SkSLThreadContext.cpp [hunk 3]
```diff
@@ -15,6 +15,12 @@
 #include "src/sksl/SkSLCompiler.h"
 #include "src/sksl/ir/SkSLExternalFunction.h"
 
+#if defined(STARBOARD)
+#include "starboard/common/log.h"
+#include "starboard/once.h"
+#include "starboard/thread.h"
+#endif
+
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply third_party/skia/src/sksl/SkSLThreadContext.cpp:23]: Because previously it was in the wrong namespace `SkSL`.  `#include <foo>` is never okay to put in a local namespace, it breaks things significantly.

It only worked by accident before.
